### PR TITLE
Add eslint for JS linting

### PR DESCRIPTION
**Pull request by sbc100**
I chose a single one of the compiler JS files and made it conform
to the style.

We use JS is strange ways (use use eval to import modules into
the global scope!) so we can't apply some rules.

Also we can't lint library.js files because they use the preprocessor.

### CODE REVIEW

**Comment by curiousdannii** [comment]
Could you add the command as a script in package.json to make it easier to run before committing? The CI scripts can then run `npm run lint`.

**Comment by sbc100** [comment]
> Could you add the command as a script in package.json to make it easier to run before committing? The CI scripts can then run `npm run lint`.

Good idea.  Thanks for the tip.  I'm not really node/npm person so I'm still learning how this stuff works.

**Comment by kripken** [comment]
How much of the changes to `parseTools.js` was automated?

**Comment by sbc100** [comment]
> How much of the changes to `parseTools.js` was automated?

I made the changes manually... even though eslint has a `--fix` I though it best to do it myself.   If you trust tools more (and/or doubt we have sufficient CI coverage) I can re-do with the automatic approach and see if it matches.

**Comment by sbc100** [comment]
> > How much of the changes to `parseTools.js` was automated?
> 
> I made the changes manually... even though eslint has a `--fix` I though it best to do it myself. If you trust tools more (and/or doubt we have sufficient CI coverage) I can re-do with the automatic approach and see if it matches.

There are enough another cleanups here that I'd like to stick with this change as is.   I could split it out into those that come from the automatic `eslint --fix` and those that required manual fixing, but I don't think its worth it.   The cleanups are most straight forward I think and we should have enough test coverage of this code.

**Comment by sbc100** [comment]
ping..

**Comment by kripken** [comment]
The cleanups seem good to me. My only concern was whether we are considering something automatic like that for the generated code too. That would be trickier for code size reasons.

File: .eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+env:
+  browser: true
+  es2020: true
+  node: true
+extends:
+  - google
+parserOptions:
+  ecmaVersion: 12
+ignorePatterns: "library_*.js"
```
Comment (kripken) [inline .eslintrc.yml:9]: is this meant to ignore the generated code? then there are also pre/postamble e tc.

**Review by kripken — APPROVED** [review]

File: .eslintrc.yml (hunk 1 above)
Comment (sbc100) [reply .eslintrc.yml:9]: So far I'm just running this on a single file.. so this pattern is kind of redundant for now.

**Comment by sbc100** [comment]
The initial goal is to run this only on built tools, not on generated code.      I don't see any reasonable way to run it on the code that we preprocess, plus we have different JS version requirements there.

**Pull request merged**